src/providers/DocumentContentProvider.ts

```diff
@@ -142,7 +142,7 @@ export class DocumentContentProvider implements vscode.TextDocumentContentProvid
           query: `ns=${namespace}`,
         });
       }
-    } else if (isCsp) {
+    } else if (isCsp && !params.has("csp")) {
```

Comment:
Also seems unnecessary as above, though in this case maybe harmless.

Reply:
This is the one that's necessary for this PR. Otherwise the generated URI won't match the file system and the user could potentially open two copies of the document at the same time.